index.js

```diff
@@ -32,22 +30,31 @@ class KeyState extends q.DesktopApp {
   // eslint-disable-next-line class-methods-use-this
   async run() {
     KeyState.log('Running...');
+
+    this.pollingInterval = this.config.pollingInterval || 1000;
+    this.colors = {
+      on: this.config.colorOn || '#FF0000',
+      off: this.config.colorOff || '#00FF00',
+    };
     const key = this.config.key || this.keys.CapsLock.code;
-    return KeyState.getKeyStatus(key)
-      .then((status) => {
-        const enabled = status === 'True';
-        return new q.Signal({
-          points: [
-            [this.generatePoint(enabled)],
-          ],
-          name: `${key} State`,
-          message: `${key} is ${enabled ? 'enabled' : 'disabled'}`,
-        });
-      }).catch((err) => {
-        // TODO: Introduce an error signal
-        // https://www.daskeyboard.io/applet-development/signals#signal-options
-        KeyState.log(err, 'error');
-      });
+
+    try {
+      var enabled = cmk.getModifierState(key);
+      KeyState.log(`${key} is ${enabled ? 'enabled' : 'disabled'}`);
+      return new q.Signal({
+        points: [
+          [this.generatePoint(enabled)],
+        ],
+        name: `${key} State`,
+        message: `${key} is ${enabled ? 'enabled' : 'disabled'}`,
+      });    
+    } catch(err){
+      KeyState.log('error');
```

Comment:
I think we should continue to log out the error as in removed line 49 above.

Reply:
Agreed - I didn't realize that I had changed that.